File: Problem_List/02433.py
class Solution:
    def findArray(self, pref: list[int]) -> list[int]:
        n = len(pref)
        arr = [0]*n
        arr[0] = pref[0]
        prefix_arr = arr[0]
        for i in range(1,n):
            arr[i] = pref[i]^prefix_arr
            prefix_arr ^= arr[i]
        return arr

File: ProblemList/02433.cpp
#include <vector>
using namespace std;
class Solution {
public:
    vector<int> findArray(vector<int>& pref) {
        vector<int> ans(pref.size(),0);
        ans[0] = pref[0];
        int cur = pref[0];
        for(int i=1; i<pref.size(); i++){
            ans[i] = cur ^ pref[i];
            cur ^= ans[i];
        }
        return ans;
    }
};